src/spawner/spawner.py

```diff
@@ -52,37 +54,103 @@ def _make(self, data: dict) -> "SpawnerConfigurationXSchedule":
     schedule_configuration_id = ForeignKeyField(ScheduleConfiguration, null=False)
 
 
-class Spawner(Component):
-    """Abstract class for spawners. Spawners can spawn SUMO vehicles
-    based on shedules.
-    """
+class ISpawnerDisruptor(ABC):
+    """Interface for the FaultInjector to block and unblock schedules"""
 
-    @classmethod
     @abstractmethod
-    def from_json(cls, json_object: str) -> "Spawner":
-        """Constructs a spawner from a JSON object.
+    def block_schedule(self, schedule_id: str):
+        """Blocks a schedule.
 
-        :param json_object: The JSON object.
-        :type json_object: str
-        :return: A spawner.
-        :rtype: Spawner
+        :param schedule_id: The id of the schedule to block
         """
         raise NotImplementedError()
 
     @abstractmethod
-    def add_schedules_from_json(self, json_schedules: list[str]):
-        """Adds schedules to the spawner from a list of JSON objects.
+    def unblock_schedule(self, schedule_id: str):
+        """Unblocks a schedule.
 
-        :param json_schedules: The list of JSON objects.
-        :type json_schedules: list[str]
+        :param schedule_id: The id of the schedule to unblock
         """
         raise NotImplementedError()
 
-    @abstractmethod
-    def spawn_specific(self, schedule: Schedule):
-        """Spawns vehicles based on a schedule.
 
-        :param schedule: The schedule.
-        :type schedule: Schedule
+class Spawner(Component, ISpawnerDisruptor):
+    """Abstract class for spawners. Spawners can spawn SUMO vehicles
+    based on shedules.
+    """
+
+    configuration: SpawnerConfiguration
+    traci_wrapper: "ITraCiWrapper"
```

Comment:
You should maybe change this type to something else (the updating component, the Train class, ...)